nautobot/core/graphql/schema.py

```diff
@@ -425,36 +403,65 @@ def already_present(model):
 
             type_identifier = f"{app_name}.{model_name}"
 
-            if type_identifier in registry["graphql_types"].keys():
+            if type_identifier in reg:
                 # Skip models that have been added statically
                 continue
 
             schema_type = generate_schema_type(app_name=app_name, model=model)
-            registry["graphql_types"][type_identifier] = schema_type
+            reg[type_identifier] = schema_type
+
+    return reg
+
+
+def generate_query_mixin():
+    """Generates and returns a class definition representing a GraphQL schema."""
+
+    class_attrs = {}
+    reg = make_reg()
+
+    def already_present(model):
+        """Check if a model and its resolvers are staged to added to the Mixin."""
+
+        single_item_name = str_to_var_name(model._meta.verbose_name)
+        list_name = str_to_var_name(model._meta.verbose_name_plural)
+
+        if single_item_name in class_attrs:
+            logger.warning(
+                f"Unable to register the schema single type '{single_item_name}' in GraphQL, "
+                f"there is already another type {class_attrs[single_item_name]._type} registered under this name"
+            )
+            return True
+
+        if list_name in class_attrs:
+            logger.warning(
+                f"Unable to register the schema list type '{list_name}' in GraphQL, "
+                f"there is already another type {class_attrs[list_name]._type} registered under this name"
+            )
+            return True
 
     # Add all objects in the plugin registry to the main registry
     # After checking for conflict
     for schema_type in registry["plugin_graphql_types"]:
         model = schema_type._meta.model
         type_identifier = f"{model._meta.app_label}.{model._meta.model_name}"
 
-        if type_identifier in registry["graphql_types"]:
+        if type_identifier in reg:
             logger.warning(
                 f'Unable to load schema type for the model "{type_identifier}" as there is already another type '
                 "registered under this name. If you are seeing this message during plugin development, check to "
                 "make sure that you aren't using @extras_features(\"graphql\") on the same model you're also "
                 "defining a custom GraphQL type for."
             )
         else:
-            registry["graphql_types"][type_identifier] = schema_type
+            reg[type_identifier] = schema_type
```

Comment:
Would this need to use `make_class()` to properly handle plugin-defined custom types as well?

Though if so, we need to think about how to make this backwards-compatible with existing plugins that are using `DjangoObjectType` instead of `NautobotObjectType`...